packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -33,25 +29,32 @@ const AvatarGroupRoot = styled('div', {
     [`& .${avatarGroupClasses.avatar}`]: styles.avatar,
     ...styles.root,
   }),
-})(({ theme, ownerState }) => {
-  const marginValue =
-    ownerState.spacing && SPACINGS[ownerState.spacing] !== undefined
-      ? SPACINGS[ownerState.spacing]
-      : -ownerState.spacing;
-
-  return {
-    [`& .${avatarClasses.root}`]: {
-      border: `2px solid ${(theme.vars || theme).palette.background.default}`,
-      boxSizing: 'content-box',
-      marginLeft: marginValue ?? -8,
-      '&:last-child': {
-        marginLeft: 0,
+})(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'row-reverse',
+  [`& .${avatarClasses.root}`]: {
+    border: `2px solid ${(theme.vars || theme).palette.background.default}`,
+    boxSizing: 'content-box',
+  },
+  variants: [
+    {
```

Comment:
This one is tricky because the `ownerState.spacing` can be any number. This falls into the ["styling based on runtime"](https://github.com/mui/material-ui/tree/master/packages/pigment-css-react#styling-based-on-runtime-values).

Here is the tricky part, we cannot use callback syntax in this PR (and all the Material UI components) because emotion does not support the callback yet (my [PR](https://github.com/emotion-js/emotion/pull/3146) has not been merged for a while), so we need to create a CSS variable and apply the inline style directly.

```suggestion
    marginLeft: 'var(--AvatarGroup-spacing, -8px)',
    '&:last-child': {
      marginLeft: 0,
    }
  },
```

and then in the component implementation:

```js
   const marginValue =
    ownerState.spacing && SPACINGS[ownerState.spacing] !== undefined
      ? SPACINGS[ownerState.spacing]
      : -ownerState.spacing;
   return (
      <AvatarGroupRoot
        as={component}
        ownerState={ownerState}
        className={clsx(classes.root, className)}
        ref={ref}
        {...other}
        style={{
          '--Avatar-spacing': marginValue ? `-${marginValue}px` : undefined,
          …other.style,
        }}
      >
```